apikey-service/src/main/java/eu/europeana/apikey/controller/ApikeyController.java

```
@@ -116,43 +120,27 @@ public ApikeyController(ApikeyRepo apikeyRepo) {
-                                   apikeyCreate.getEmail(),
-                                   null != apikeyCreate.getLevel() && apikeyCreate.getLevel()
-                                            .equalsIgnoreCase(Level.ADMIN.getLevelName()) ?
-                                            Level.ADMIN.getLevelName() : Level.CLIENT.getLevelName());
-        if (null != apikeyCreate.getWebsite()) {
-            apikey.setWebsite(apikeyCreate.getWebsite());
-        }
-        if (null != apikeyCreate.getAppName()) {
-            apikey.setAppName(apikeyCreate.getAppName());
-        }
-        if (null != apikeyCreate.getCompany()) {
-            apikey.setCompany(apikeyCreate.getCompany());
-        }
-        if (null != apikeyCreate.getSector()) {
-            apikey.setSector(apikeyCreate.getSector());
+        KeycloakAuthenticationToken keycloakAuthenticationToken = (KeycloakAuthenticationToken) SecurityContextHolder.getContext().getAuthentication();
+        KeycloakSecurityContext securityContext = (KeycloakSecurityContext) keycloakAuthenticationToken.getCredentials();
+        try {
+            FullApikey apikey = keycloakManager.createClient(securityContext, apikeyCreate);
+            if (apikey != null) {
+                this.apikeyRepo.save(new Apikey(apikey));
+                LOG.debug("apikey: {} created", apikey.getApikey());
+
+                emailService.sendSimpleMessageUsingTemplate(apikey.getEmail(),
+                        "Your Europeana API keys",
+                        apikeyCreatedMail,
+                        apikey.getFirstName(),
+                        apikey.getLastName(),
+                        apikey.getApikey(),
+                        apikey.getClientSecret());
+                return new ResponseEntity<>(apikey, HttpStatus.CREATED);
+            }
+        } catch (ApikeyException e) {
+            return new ResponseEntity<>(e, HttpStatus.valueOf(e.getStatus()));
```

the error response should return an error message in the body